lib/cooked_post_processor.rb

```diff
@@ -339,12 +339,7 @@ def convert_to_link!(img)
       width, height = ImageSizer.resize(width, height)
     end
 
-    # if the upload already exists and is attached to a different post,
-    # or the original_sha1 is missing meaning it was created before secure
-    # media was enabled. we want to re-thumbnail and re-optimize in this case
-    # to avoid using uploads linked to many other posts
-    upload = Upload.consider_for_reuse(Upload.get_from_url(src), @post)
-
```

Comment:
This is a nice change. I agree there is no need to prevent thumbnail generation for copied secure uploads 👍